### PR TITLE
feat(FX-4756): Save updated artwork lists for artwork

### PR DESCRIPTION
**Pull request by dimatretyak**
This PR resolves [FX-4756] 

#### iOS
https://user-images.githubusercontent.com/3513494/234855215-1d9952b3-4ec1-42ab-8e83-4b2c4b817b59.mp4

#### Android
https://user-images.githubusercontent.com/3513494/234858040-e06bc5d7-d41d-4b21-8738-d88b44240b9b.mp4

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [ ] I added an **[app state migration]**, or my changes do not require one.
- [ ] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Save updated artwork lists for artwork - dimatretyak

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[FX-4756]: https://artsyproduct.atlassian.net/browse/FX-4756?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Components/ArtworkLists/views/SelectArtworkListsForArtworkView/components/SelectArtworkListsForArtwork.tsx [hunk 1]
```diff
@@ -12,9 +12,15 @@ const LOADING_PLACEHOLDER_COUNT = 10
...
-    artworkID: artwork.internalID,
-  })
+  const queryData = useLazyLoadQuery<SelectArtworkListsForArtworkQuery>(
+    Query,
+    {
+      artworkID: artwork.internalID,
+    },
+    {
+      fetchPolicy: "network-only",
+    }
```
Comment (dimatretyak) [inline src/app/Components/ArtworkLists/views/SelectArtworkListsForArtworkView/components/SelectArtworkListsForArtwork.tsx:22]: always query up-to-date data and ignore cached

File: src/app/Components/ArtworkLists/views/SelectArtworkListsForArtworkView/components/SelectArtworkListsForArtworkFooter.tsx [hunk 2]
```diff
@@ -1,17 +1,40 @@
...
 import { ArtworkListsViewName } from "app/Components/ArtworkLists/views/constants"
 
 export const SelectArtworkListsForArtworkFooter = () => {
   const { state } = useArtworkListsContext()
   const { dismiss } = useBottomSheetModal()
   const { addingArtworkListIDs, removingArtworkListIDs, selectedArtworkListIDs } = state
   const hasChanges = addingArtworkListIDs.length !== 0 || removingArtworkListIDs.length !== 0
+  const artwork = state.artwork!
+
+  const [commit, mutationInProgress] = useUpdateArtworkListsForArtwork(artwork.id)
```
Comment (dimatretyak) [inline src/app/Components/ArtworkLists/views/SelectArtworkListsForArtworkView/components/SelectArtworkListsForArtworkFooter.tsx:15]: **question**: do you think it's a good idea to store the `updater' inside a custom hook, or is it better to [prev approach](https://github.com/artsy/eigen/pull/8585/commits/9c25902a25133792790aaf9bcdcc92dabfc2d85d)? 🤔

Comment (dblandin) [reply src/app/Components/ArtworkLists/views/SelectArtworkListsForArtworkView/components/SelectArtworkListsForArtworkFooter.tsx:15]: I don't have a strong opinion but seems nice to have the updater co-located with some of the other relay store logic within the custom hook. /shrug

**Review by dblandin — APPROVED** [review]
Nice! Looking good! 💯

**Pull request merged**